iam_builder/templates.py

```diff
@@ -98,6 +98,100 @@
+                "glue:DeleteDatabase",
+                "glue:UpdateTable",
+                "glue:UpdateSchema",
+                "glue:UpdatePartition",
+                "glue:UpdateDatabase",
+                "glue:CreateTable",
+                "glue:CreateSchema",
+                "glue:CreatePartition",
+                "glue:CreatePartitionIndex",
+                "glue:BatchCreatePartition",
+                "glue:CreateDatabase"
+            ],
+            "Resource": [
+                "arn:aws:glue:*:*:schema/*",
+                "arn:aws:glue:*:*:database/*",
+                "arn:aws:glue:*:*:table/*/*",
+                "arn:aws:glue:*:*:catalog"
+            ]
+        },
+        {
+            "Sid": "AthenaActions",
+            "Effect": "Allow",
+            "Action": [
+                "athena:List*",
+                "athena:Get*",
+                "athena:StartQueryExecution",
+                "athena:StopQueryExecution"
+            ],
+            "Resource": [
+                "arn:aws:athena:*:*:datacatalog/*",
+                "arn:aws:athena:*:*:workgroup/*"
+            ]
+        },
+        {
+            "Sid": "AirflowCLIPolicy",
+            "Effect": "Allow",
+            "Action": [
+                "airflow:CreateCliToken"
+            ],
+            "Resource": [
+                "arn:aws:airflow:*:*:environment/dev",
+                "arn:aws:airflow:*:*:environment/prod"
+            ]
+        },
+        {
+            "Sid": "CadetWriteAccess",
+            "Effect": "Allow",
+            "Action": [
+                "s3:List*",
+                "s3:GetObject*",
+                "s3:GetBucket*",
+                "s3:DeleteObject*",
+                "s3:PutObject*"
+            ],
+            "Resource": [
+                "arn:aws:s3:::mojap-derived-tables/*",
+                "arn:aws:s3:::mojap-derived-tables",
+                "arn:aws:s3:::dbt-query-dump/*",
+                "arn:aws:s3:::dbt-query-dump",
+                "arn:aws:s3:::mojap-manage-offences/ho-offence-codes/*",
+                "arn:aws:s3:::mojap-manage-offences",
+                "arn:aws:s3:::mojap-hub-exports/probation_referrals_dump/*",
+                "arn:aws:s3:::mojap-hub-exports",
+                "arn:aws:s3:::alpha-app-opg-lpa-dashboard",
+                "arn:aws:s3:::alpha-app-opg-lpa-dashboard/dev/models/domain_name=opg/*",
+                "arn:aws:s3:::alpha-app-opg-lpa-dashboard/prod/models/domain_name=opg/*",
+                "arn:aws:s3:::alpha-bold-data-shares",
+                "arn:aws:s3:::alpha-bold-data-shares/reducing-reoffending/*"
+            ]
+        },
+        {
+            "Sid": "CadetReadAccess",
+            "Effect": "Allow",
+            "Action": [
+                "s3:List*",
+                "s3:GetObject*",
+                "s3:GetBucket*"
+            ],
+            "Resource": [
+                "arn:aws:s3:::*",
+                "arn:aws:s3:::*/*"
```

Comment:
Again for my understanding, why are both of these required? Would `arn:aws:s3:::*` not just be a super set of the one below it?

Reply:
This is because the overall permission technically covers two different types of resources - Buckets, which have a resource type of `"arn:aws:s3:::<bucket-name>"`, and then the objects within those buckets, which have a resource type of `"arn:aws:s3:::<bucket-name>/<some-path>". Where we have a blended statement like this, AWS will intelligently only try and apply bucket permissions to bucket looking resources, and vice versa for object looking permissions. Given we need to cover essentially all buckets and all object within those buckets as potential data sources for CaDeT, we therefore need both statements, and AWS will resolve which permissions apply to which resources on the back end.